### PR TITLE
copr: Refine `EvalContext::handle_overflow`

### PR DESCRIPTION
**Pull request by iosmanthus**
Signed-off-by: Iosmanthus Teng <myosmanthustree@gmail.com>

## What have you changed? (mandatory)
`handle_overflow` should be consistent with `handle_truncate`:
```rust
    pub fn handle_truncate(&mut self, is_truncated: bool) -> Result<()> {
        if !is_truncated {
            return Ok(());
        }
        self.handle_truncate_err(Error::truncated())
    }

    pub fn handle_truncate_err(&mut self, err: Error) -> Result<()> {
        if self.cfg.flag.contains(Flag::IGNORE_TRUNCATE) {
            return Ok(());
        }
        if self.cfg.flag.contains(Flag::TRUNCATE_AS_WARNING) {
            self.warnings.append_warning(err);
            return Ok(());
        }
        Err(err)
    }

```

## What are the type of changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:
- Improvement (a change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)
`cargo check --all-targets`

### CODE REVIEW

File: components/tidb_query/src/expr/ctx.rs [hunk 1]
```diff
@@ -239,9 +239,17 @@ impl EvalContext {
         Err(err)
     }
 
+    pub fn handle_overflow(&mut self, is_overflow: bool) -> Result<()> {
```
Comment (sticnarf) [inline components/tidb_query/src/expr/ctx.rs]: Looks like it's used nowhere. :/

Comment (iosmanthus) [reply components/tidb_query/src/expr/ctx.rs]: I need it in another PR. :laughing:

Comment (sticnarf) [reply components/tidb_query/src/expr/ctx.rs]: I think you can combine them. This PR is not too big.

**Pull request merged**